Add libunwind

kraileth opened the pull request:
The change to ravensys-toolchain was necessary to make libunwind build successfully (see thread on ravenports-devel).

The port for libunwind builds and passes the tests, however currently the architecture is hard-coded as I don't know what Ravenadm offers to extract the architecture from. How should this be solved?

jrmarino commented:
The change to ravensys-toolchain is an absolute no-go.

jrmarino commented:
the ravenports-devel thread was missing a reply.

What is libunwind proposed for?
It doesn't build on DF, nor is it's suitability on solaris known.  Anything that *requires* it will be inaccessible, and everything downstream of that.  So it's good to know the options here.

kraileth commented:
I'm still trying to port webkit2. Every port I wrote for some time now is meant to support it. Currently webkit is asking for gstreamer - however not the already ported version 0.10. The port for gstreamer1 on FreeBSD lists libunwind as a dependency. Looking at dports, though, it looks like it can simply be dropped? If that's the case this pull request should probably just be cancelled...

jrmarino commented:
freebsd is extremely careless wrt to libunwind.  It was (unnecessarily) added to ruby and as a result, a lot of rubygems added too, thinking it was necessary (or maybe earlier it was due to linking issues).

The real question that should be asked when seen is: "Why is libunwind needed?"
Generally unwinding comes with gcc.  or maybe it's used for debugging capability.  
In any case, let's investigate each requirement to ascertain if it's really required.  I have a feeling a lot of these are "thrown in".

jrmarino commented:
that said, i don't object to have a libunwind port -- it might be useful as a standalone library of separate projects.

kraileth commented:
Ok, thanks for explaining. Since your answers on the mailing list were not visible, I tried to find a solution on my own.

I've reverted the ravensys-toolchain changes and added changed the specfile for libunwind as you suggested (and indicated to only build it for FreeBSD and Linux - or would you prefer to only disable it for DF since it's not known if it works on Solaris?). Does Ravenadm have any facility where I can grab the architecture from? If yes, that would be something that makes sense using, too.

jrmarino commented:
when I replied to the ravenports-devel email, I should have "replied to all".  So the email still should have gone to your account, but the mail list didn't get a copy.  Anyway ,,, it's fine like it is.  When I get a chance (and probably acquire a better build machine) I can check it on solaris.

jrmarino commented:
the freebsd patch seems like it should be generic (iow, in the "patches" directory)

kraileth commented:
Done.
